packages/nuxt/src/core/nitro.ts

```diff
@@ -366,6 +366,13 @@ export async function initNitro (nuxt: Nuxt & { _nitro?: Nitro }) {
   // Init nitro
   const nitro = await createNitro(nitroConfig)
 
+  // Trigger Nitro reload when SPA loading template changes
+  nuxt.hook('builder:watch', (_event, path) => {
+    if (normalize(path) === relative(nuxt.options.rootDir, resolve(nuxt.options.srcDir, 'app/spa-loading-template.html'))) {
+      nitro.hooks.callHook('rollup:reload')
```

Comment:
would be nice to chain the hooks (await) to prevent unhandled promise errors

Reply:
it will still be unhandled as we do not await that hook.